registration/include/pcl/registration/transformation_estimation_symmetric_point_to_plane_lls.h

```diff
@@ -0,0 +1,154 @@
+
+        /** \brief Estimate a rigid rotation transformation between a source and a target point cloud using SVD.
+          * \param[in] cloud_src the source point cloud dataset
+          * \param[in] indices_src the vector of indices describing the points of interest in \a cloud_src
+          * \param[in] cloud_tgt the target point cloud dataset
+          * \param[in] indices_tgt the vector of indices describing the correspondences of the interest points from \a indices_src
+          * \param[out] transformation_matrix the resultant transformation matrix
+          */
+        inline void
+        estimateRigidTransformation (
+            const pcl::PointCloud<PointSource> &cloud_src,
+            const std::vector<int> &indices_src,
+            const pcl::PointCloud<PointTarget> &cloud_tgt,
+            const std::vector<int> &indices_tgt,
+            Matrix4 &transformation_matrix) const override;
+
+        /** \brief Estimate a rigid rotation transformation between a source and a target point cloud using SVD.
+          * \param[in] cloud_src the source point cloud dataset
+          * \param[in] cloud_tgt the target point cloud dataset
+          * \param[in] correspondences the vector of correspondences between source and target point cloud
+          * \param[out] transformation_matrix the resultant transformation matrix
+          */
+        inline void
+        estimateRigidTransformation (
+            const pcl::PointCloud<PointSource> &cloud_src,
+            const pcl::PointCloud<PointTarget> &cloud_tgt,
+            const pcl::Correspondences &correspondences,
+            Matrix4 &transformation_matrix) const override;
+
+      protected:
+        
+        /** \brief Estimate a rigid rotation transformation between a source and a target
+          * \param[in] source_it an iterator over the source point cloud dataset
+          * \param[in] target_it an iterator over the target point cloud dataset
+          * \param[out] transformation_matrix the resultant transformation matrix
+          */
+        void 
+        estimateRigidTransformation (ConstCloudIterator<PointSource>& source_it, 
+                                     ConstCloudIterator<PointTarget>& target_it, 
+                                     Matrix4 &transformation_matrix) const;
+
+        /** \brief Construct a 4 by 4 transformation matrix from the provided rotation and translation.
+          * \param[in] alpha the rotation about the x-axis
+          * \param[in] beta the rotation about the y-axis
+          * \param[in] gamma the rotation about the z-axis
+          * \param[in] tx the x translation
+          * \param[in] ty the y translation
+          * \param[in] tz the z translation
+          * \param[out] transformation_matrix the resultant transformation matrix
+          */
+        inline void
+        constructTransformationMatrix (const double & alpha, const double & beta, const double & gamma,
```

Comment:
Why do you prefer a longform function parameter list? For me personally, it makes more sense looking at an Eigen::Vector3d (or double[3]) translation. Same for the euler angles. (Sorry if the angles aren't Euler. Such rotations are usually XYZ Euler, but I haven't read the paper)

Reply:
Switched to a Vector6 parameter list which is already being used by the estimateRigidTransformation caller.